packages/vest/src/core/test/__tests__/each.test.js

```diff
@@ -0,0 +1,132 @@
+import faker from 'faker';
+import enforce from 'n4s';
+
+import test from 'test';
+import vest from 'vest';
+
+let testObjects;
+
+describe("Test Vest's `test.each` function", () => {
+  describe('Sync Tests', () => {
+    describe('Basic Functionality', () => {
+      it('Should return all test objects correctly', () => {
+        vest.create(faker.random.word(), () => {
+          testObjects = test.each([[1, 2, 3], [2, 7, 9]])(faker.random.word(), faker.lorem.sentence(), (a, b, c) => {
+            enforce(a + b).equals(c);
+          });
+        })();
+        expect(testObjects).toHaveLength(2);
+        expect(testObjects[0].failed).toBe(false);
+        expect(testObjects[1].failed).toBe(false);
+      });
+
+      it('Should mark failed tests as such', () => {
+        vest.create(faker.random.word(), () => {
+          testObjects = test.each([[5, 5, 10], [1, 4, 10], [2, 7, 9]])(faker.random.word(), faker.lorem.sentence(), (a, b, c) => {
+            enforce(a + b).equals(c);
+          });
+        })();
+        expect(testObjects).toHaveLength(3);
+        expect(testObjects[0].failed).toBe(false);
+        expect(testObjects[1].failed).toBe(true);
+        expect(testObjects[2].failed).toBe(false);
+      });
+
+      it('Should include function message when fail test uses function statement', () => {
+        vest.create(faker.random.word(), () => {
+          testObjects = test.each([[5, 4, 10]])(faker.random.word(), (a, b, c) => {
+            return `${a} + ${b} != ${c}`;
+          }, (a, b, c) => {
+            enforce(a + b).equals(c);
+          });
+        })();
+        expect(testObjects).toHaveLength(1);
+        expect(testObjects[0].failed).toBe(true);
+        expect(testObjects[0].statement).toBe('5 + 4 != 10');
+      });
+
+      it('Should work correctly with exclusions', () => {
+        const statementFn1 = jest.fn();
+        const testFn1 = jest.fn();
+        const testFn2 = jest.fn();
+
+        const validate = vest.create(faker.random.word(), () => {
+          vest.only('test2');
+          test.each([[5, 4, 10]])('test1', statementFn1, testFn1);
+          test.each([[5, 4, 10]])('test2', faker.lorem.sentence(), testFn2);
+        });
+        validate();
+        expect(testFn1).not.toHaveBeenCalled();
+        expect(statementFn1).toHaveBeenCalled();
+        expect(testFn2).toHaveBeenCalled();
```

Comment:
We can also test the field against the result:

```suggestion
        const result = validate();
        expect(testFn1).not.toHaveBeenCalled();
        expect(result.tests.test1).not.toBeDefined();
        expect(statementFn1).toHaveBeenCalled();
        expect(testFn2).toHaveBeenCalled();
        expect(result.tests.test2).toBeDefined();
```